packages/backend/src/lib/git-instance.ts

```diff
@@ -105,6 +111,36 @@ export class GitInstance {
     return sha;
   }
 
+  /**
+   * Rollback to a previous commit using the git commit hash
+   * @param gitHash Commit has to rollback to
+   * @param gitUrl Git repository url from Insight
+   * @param user User making the changes
+   * @returns insight yaml from local path
+   */
+  public static async rollBackCommit({ gitHash, gitUrl, user }: RollBackCommitArgs): Promise<InsightYaml> {
+    const { displayName: name, email, githubPersonalAccessToken } = user;
+
+    const gitInstance = await GitInstance.from(gitUrl, githubPersonalAccessToken!);
+
+    // Checkout changes from commit using hash
+    logger.debug(`Checkout from commit hash to restore files`);
+    await gitInstance.checkoutOnRef(gitHash, githubPersonalAccessToken!);
+
+    // Commit changes and push
+    logger.debug(`Making commit as user ${name} (${email})`);
+    await gitInstance.commit(`Rollback changes from ${gitHash.slice(0, 7)}`, { name, email });
+    logger.debug(`Pushing change to origin!`);
+    await gitInstance.push(githubPersonalAccessToken!);
+    logger.debug(`Changes pushed successfully!`);
+    await gitInstance.cleanup();
+
+    // Get insightYaml to sync the Insight
+    const insightYaml = await gitInstance.retrieveInsightYaml();
+
+    return insightYaml;
```

Comment:
Getting `insightYaml` here to use it on the `rollBackToCommit` method from the change-history service, in order to sync the repository to the Insight.